hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiverServer.java

```diff
@@ -68,8 +68,7 @@
    * Enforcing the limit is required in order to avoid data-node
    * running out of memory.
    */
-  int maxXceiverCount =
-    DFSConfigKeys.DFS_DATANODE_MAX_RECEIVER_THREADS_DEFAULT;
+  volatile int maxXceiverCount;
```

Comment:
Looks like we need volatile for sure right? There might be slightest chance for DataXceiverServer thread to throw limit exceeded error if this is not volatile?

Reply:
Thanks @virajjasani for your review. IMO, to ensure consistency across threads, we need to make it volatile.